--- origsrc/shotwell-0.11.4/plugins/Makefile.plugin.mk	2011-10-13 14:22:54.000000000 -0500
+++ src/shotwell-0.11.4/plugins/Makefile.plugin.mk	2011-10-16 17:27:18.412498000 -0500
@@ -32,7 +32,7 @@ SRC_FILES := ../common/Resources.vala $(
 CFILES := $(notdir $(SRC_FILES:.vala=.c))
 OFILES := $(notdir $(SRC_FILES:.vala=.o))
 
-CFLAGS := `pkg-config --print-errors --cflags $(EXT_PKGS)` -nostdlib -export-dynamic $(PLUGIN_CFLAGS)
+CFLAGS := `pkg-config --print-errors --cflags $(EXT_PKGS)` $(PLUGIN_CFLAGS)
 LDFLAGS := `pkg-config --print-errors --libs $(EXT_PKGS)` $(LDFLAGS)
 DEFINES := -D_VERSION='"$(PLUGINS_VERSION)"' -DGETTEXT_PACKAGE='"shotwell"'
 
@@ -54,7 +54,7 @@ $(OFILES): %.o: %.c $(CFILES)
 	$(CC) -c $(CFLAGS) $(DEFINES) -I../.. $(CFILES)
 
 $(PLUGIN).so: $(OFILES)
-	$(CC) $(CFLAGS) -Wl,--no-as-needed $(LDFLAGS) $(OFILES) -I../.. -shared -o $@
+	$(CC) $(CFLAGS) $(OFILES) $(LDFLAGS) ../../libshotwell.a -I../.. -shared -o $@
 
 .PHONY: cleantemps
 cleantemps:
--- origsrc/shotwell-0.11.4/shotwell-plugin-dev-1.0.m4	2011-09-23 17:52:27.000000000 -0500
+++ src/shotwell-0.11.4/shotwell-plugin-dev-1.0.m4	2011-10-16 17:07:05.410384000 -0500
@@ -8,4 +8,5 @@ Description: Headers for building Shotwe
 Requires: _REQUIREMENTS_
 Version: _VERSION_
 Cflags: -I${includedir}/shotwell
+Libs: -L${libdir} -lshotwell
 
