--- origsrc/shotwell-0.11.4/Makefile	2011-10-13 14:26:06.000000000 -0500
+++ src/shotwell-0.11.4/Makefile	2011-10-16 17:05:49.219851200 -0500
@@ -273,7 +273,6 @@ EXT_PKGS = \
 	gstreamer-base-0.10 \
 	gstreamer-pbutils-0.10 \
 	gtk+-2.0 \
-	gudev-1.0 \
 	libexif \
 	libgphoto2 \
 	libsoup-2.4 \
@@ -309,7 +308,6 @@ EXT_PKG_VERSIONS = \
 	gstreamer-base-0.10 >= 0.10.28 \
 	gstreamer-pbutils-0.10 >= 0.10.32 \
 	gtk+-2.0 >= 2.18.0 \
-	gudev-1.0 >= 145 \
 	libexif >= 0.6.16 \
 	libgphoto2 >= 2.4.2 \
 	libsoup-2.4 >= 2.26.0 \
--- origsrc/shotwell-0.11.4/src/BatchImport.vala	2011-10-13 14:22:54.000000000 -0500
+++ src/shotwell-0.11.4/src/BatchImport.vala	2011-10-16 17:07:38.264041300 -0500
@@ -117,8 +117,10 @@ public enum ImportResult {
                 return ImportResult.FILE_WRITE_ERROR;
             else
                 return ImportResult.FILE_ERROR;
+#if CAMERA
         } else if (err is GPhotoError) {
             return ImportResult.CAMERA_ERROR;
+#endif
         }
         
         return default_result;
--- origsrc/shotwell-0.11.4/src/Properties.vala	2011-10-13 14:22:54.000000000 -0500
+++ src/shotwell-0.11.4/src/Properties.vala	2011-10-16 17:14:11.427125500 -0500
@@ -158,6 +158,7 @@ private class BasicProperties : Properti
 
         title = source.get_name();
         
+#if CAMERA
         if (source is PhotoSource || source is PhotoImportSource) {           
             start_time = (source is PhotoSource) ? ((PhotoSource) source).get_exposure_time() :
                 ((PhotoImportSource) source).get_exposure_time();
@@ -165,6 +166,13 @@ private class BasicProperties : Properti
                         
             PhotoMetadata? metadata = (source is PhotoSource) ? ((PhotoSource) source).get_metadata() :
                 ((PhotoImportSource) source).get_metadata();
+#else
+        if (source is PhotoSource) {
+            start_time = ((PhotoSource) source).get_exposure_time();
+            end_time = start_time;
+                        
+            PhotoMetadata? metadata = ((PhotoSource) source).get_metadata();
+#endif
 
             if (metadata != null) {
                 exposure = metadata.get_exposure_string();
@@ -184,7 +191,9 @@ private class BasicProperties : Properti
                     Dimensions(0, 0);
             }
             
+#if CAMERA
             if (source is PhotoSource)
+#endif
                 dimensions = ((PhotoSource) source).get_dimensions();
             
             if (source is Photo && ((Photo) source).get_master_file_format() == PhotoFileFormat.RAW) {
@@ -205,8 +214,12 @@ private class BasicProperties : Properti
             
             photo_count = event_photo_count;
             video_count = event_video_count;
+#if CAMERA
         } else if (source is VideoSource || source is VideoImportSource) {
             if (source is VideoSource) {
+#else
+        } else if (source is VideoSource) {
+#endif
                 Video video = (Video) source;
                 clip_duration = video.get_clip_duration();
 
@@ -214,9 +227,11 @@ private class BasicProperties : Properti
                     dimensions = video.get_frame_dimensions();
 
                 start_time = video.get_exposure_time();
+#if CAMERA
             } else {
                 start_time = ((VideoImportSource) source).get_exposure_time();
             }
+#endif
             end_time = start_time;
         }
     }
@@ -229,10 +244,15 @@ private class BasicProperties : Properti
         foreach (DataView view in iter) {
             DataSource source = view.get_source();
             
+#if CAMERA
             if (source is PhotoSource || source is PhotoImportSource) {                  
                 time_t exposure_time = (source is PhotoSource) ?
                     ((PhotoSource) source).get_exposure_time() :
                     ((PhotoImportSource) source).get_exposure_time();
+#else
+            if (source is PhotoSource) {
+                time_t exposure_time = ((PhotoSource) source).get_exposure_time();
+#endif
 
                 if (exposure_time != 0) {
                     if (start_time == 0 || exposure_time < start_time)
@@ -268,10 +288,15 @@ private class BasicProperties : Properti
                 photo_count += event_photo_count;
                 video_count += event_video_count;
                 event_count++;
+#if CAMERA
             } else if (source is VideoSource || source is VideoImportSource) {
                 time_t exposure_time = (source is VideoSource) ?
                     ((VideoSource) source).get_exposure_time() :
                     ((VideoImportSource) source).get_exposure_time();
+#else
+            } else if (source is VideoSource) {
+                time_t exposure_time = ((VideoSource) source).get_exposure_time();
+#endif
 
                 if (exposure_time != 0) {
                     if (start_time == 0 || exposure_time < start_time)
--- origsrc/shotwell-0.11.4/src/SearchFilter.vala	2011-10-13 14:22:54.000000000 -0500
+++ src/shotwell-0.11.4/src/SearchFilter.vala	2011-10-16 17:15:30.441263300 -0500
@@ -493,8 +493,10 @@ public class SearchFilterActions {
             Core.ViewTracker? tracker = old_tracked_page.get_view_tracker();
             if (tracker is MediaViewTracker)
                 tracker.updated.disconnect(on_media_tracker_updated);
+#if CAMERA
             else if (tracker is CameraViewTracker)
                 tracker.updated.disconnect(on_camera_tracker_updated);
+#endif
         }
         
         CheckerboardPage? new_tracked_page = new_page as CheckerboardPage;
@@ -505,11 +507,13 @@ public class SearchFilterActions {
                 on_media_tracker_updated(tracker);
                 
                 return;
+#if CAMERA
             } else if (tracker is CameraViewTracker) {
                 tracker.updated.connect(on_camera_tracker_updated);
                 on_camera_tracker_updated(tracker);
                 
                 return;
+#endif
             }
         }
         
@@ -533,6 +537,7 @@ public class SearchFilterActions {
         update_sensitivities();
     }
     
+#if CAMERA
     private void on_camera_tracker_updated(Core.Tracker t) {
         CameraViewTracker tracker = (CameraViewTracker) t;
         
@@ -543,6 +548,7 @@ public class SearchFilterActions {
 
         update_sensitivities();
     }
+#endif
     
     private void update_sensitivities() {
         flagged.set_stock_id(((SearchFilterCriteria.FLAG & criteria) != 0 && has_flagged) ?
--- origsrc/shotwell-0.11.4/src/camera/mk/camera.mk	2011-09-23 17:52:27.000000000 -0500
+++ src/shotwell-0.11.4/src/camera/mk/camera.mk	2011-10-16 17:15:41.891683300 -0500
@@ -12,7 +12,6 @@ UNIT_DIR := camera
 # NOTE: Do *not* include the unit's master file, i.e. UNIT_NAME.vala.
 UNIT_FILES := \
 	Branch.vala \
-	CameraTable.vala \
 	GPhoto.vala \
 	ImportPage.vala
 
--- origsrc/shotwell-0.11.4/src/library/LibraryWindow.vala	2011-10-13 14:22:54.000000000 -0500
+++ src/shotwell-0.11.4/src/library/LibraryWindow.vala	2011-10-16 17:17:31.871875000 -0500
@@ -160,7 +160,9 @@ public class LibraryWindow : AppWindow {
     private Library.FlaggedBranch flagged_branch = new Library.FlaggedBranch();
     private Library.LastImportBranch last_import_branch = new Library.LastImportBranch();
     private Library.ImportQueueBranch import_queue_branch = new Library.ImportQueueBranch();
+#if CAMERA
     private Camera.Branch camera_branch = new Camera.Branch();
+#endif
     private Searches.Branch saved_search_branch = new Searches.Branch();
     
     private Gee.HashMap<Page, Sidebar.Entry> page_map = new Gee.HashMap<Page, Sidebar.Entry>();
@@ -215,7 +217,9 @@ public class LibraryWindow : AppWindow {
         sidebar_tree.graft(flagged_branch, SidebarRootPosition.FLAGGED);
         sidebar_tree.graft(last_import_branch, SidebarRootPosition.LAST_IMPORTED);
         sidebar_tree.graft(import_queue_branch, SidebarRootPosition.IMPORT_QUEUE);
+#if CAMERA
         sidebar_tree.graft(camera_branch, SidebarRootPosition.CAMERAS);
+#endif
         sidebar_tree.graft(saved_search_branch, SidebarRootPosition.SAVED_SEARCH);
         
         // create and connect extended properties window
@@ -260,7 +264,9 @@ public class LibraryWindow : AppWindow {
         LibraryMonitorPool.get_instance().monitor_installed.connect(on_library_monitor_installed);
         LibraryMonitorPool.get_instance().monitor_destroyed.connect(on_library_monitor_destroyed);
         
+#if CAMERA
         CameraTable.get_instance().camera_added.connect(on_camera_added);
+#endif
     }
     
     ~LibraryWindow() {
@@ -288,7 +294,9 @@ public class LibraryWindow : AppWindow {
         LibraryMonitorPool.get_instance().monitor_installed.disconnect(on_library_monitor_installed);
         LibraryMonitorPool.get_instance().monitor_destroyed.disconnect(on_library_monitor_destroyed);
         
+#if CAMERA
         CameraTable.get_instance().camera_added.disconnect(on_camera_added);
+#endif
     }
     
     private void on_library_monitor_installed(LibraryMonitor monitor) {
@@ -1027,6 +1035,7 @@ public class LibraryWindow : AppWindow {
         switch_to_page(import_queue_branch.get_queue_page());
     }
     
+#if CAMERA
     private void on_camera_added(DiscoveredCamera camera) {
         // if this page is for a camera which initialized the app, we want to switch to that page
         if (!initial_camera_uris.contains(camera.uri))
@@ -1057,6 +1066,7 @@ public class LibraryWindow : AppWindow {
             switch_to_page(page);
         }
     }
+#endif
     
     private PageLayout? get_page_layout(Page page) {
         return page_layouts.get(page);
@@ -1536,8 +1546,10 @@ public class LibraryWindow : AppWindow {
         
         // convert file: URIs into gphoto disk: URIs
         string alt_uri = null;
+#if CAMERA
         if (uri.has_prefix("file://"))
             alt_uri = CameraTable.get_port_uri(uri.replace("file://", "disk:"));
+#endif
         
         // we only add uris when the notification is called on startup
         if (at_startup) {
--- origsrc/shotwell-0.11.4/src/library/mk/library.mk	2011-09-23 17:52:27.000000000 -0500
+++ src/shotwell-0.11.4/src/library/mk/library.mk	2011-10-16 17:17:45.225498300 -0500
@@ -42,7 +42,6 @@ UNIT_USES := \
 	Sidebar \
 	Events \
 	Tags \
-	Camera \
 	Searches \
 	Faces
 
--- origsrc/shotwell-0.11.4/units.mk	2011-09-23 17:52:27.000000000 -0500
+++ src/shotwell-0.11.4/units.mk	2011-10-16 17:17:59.343522900 -0500
@@ -24,7 +24,6 @@ UNITS = \
 	events \
 	tags \
 	faces \
-	camera \
 	searches \
 	config
 
