--- origsrc/shotwell-0.12.3/Makefile	2012-05-09 16:09:15.000000000 -0500
+++ src/shotwell-0.12.3/Makefile	2012-11-01 22:23:52.131353300 -0500
@@ -280,7 +280,6 @@ EXT_PKGS = \
 	gstreamer-base-0.10 \
 	gstreamer-pbutils-0.10 \
 	gtk+-3.0 \
-	gudev-1.0 \
 	libexif \
 	libgphoto2 \
 	libsoup-2.4 \
@@ -316,7 +315,6 @@ EXT_PKG_VERSIONS = \
 	gstreamer-plugins-base-0.10 >= 0.10.32 \
 	gstreamer-pbutils-0.10 >= 0.10.32 \
 	gtk+-3.0 >= 3.0.11 \
-	gudev-1.0 >= 145 \
 	libexif >= 0.6.16 \
 	libgphoto2 >= 2.4.2 \
 	libsoup-2.4 >= 2.26.0 \
--- origsrc/shotwell-0.12.3/src/BatchImport.vala	2012-05-08 12:42:59.000000000 -0500
+++ src/shotwell-0.12.3/src/BatchImport.vala	2012-11-01 22:23:52.131353300 -0500
@@ -117,8 +117,10 @@ public enum ImportResult {
                 return ImportResult.FILE_WRITE_ERROR;
             else
                 return ImportResult.FILE_ERROR;
+#if CAMERA
         } else if (err is GPhotoError) {
             return ImportResult.CAMERA_ERROR;
+#endif
         }
         
         return default_result;
--- origsrc/shotwell-0.12.3/src/Properties.vala	2012-05-08 12:42:59.000000000 -0500
+++ src/shotwell-0.12.3/src/Properties.vala	2012-11-01 22:23:52.131353300 -0500
@@ -158,6 +158,7 @@ private class BasicProperties : Properti
 
         title = source.get_name();
         
+#if CAMERA
         if (source is PhotoSource || source is PhotoImportSource) {           
             start_time = (source is PhotoSource) ? ((PhotoSource) source).get_exposure_time() :
                 ((PhotoImportSource) source).get_exposure_time();
@@ -165,6 +166,13 @@ private class BasicProperties : Properti
                         
             PhotoMetadata? metadata = (source is PhotoSource) ? ((PhotoSource) source).get_metadata() :
                 ((PhotoImportSource) source).get_metadata();
+#else
+        if (source is PhotoSource) {
+            start_time = ((PhotoSource) source).get_exposure_time();
+            end_time = start_time;
+                        
+            PhotoMetadata? metadata = ((PhotoSource) source).get_metadata();
+#endif
 
             if (metadata != null) {
                 exposure = metadata.get_exposure_string();
@@ -184,8 +192,10 @@ private class BasicProperties : Properti
                     Dimensions(0, 0);
             }
             
+#if CAMERA
             if (source is PhotoSource)
                 dimensions = ((PhotoSource) source).get_dimensions();
+#endif
             
             if (source is Photo && ((Photo) source).get_master_file_format() == PhotoFileFormat.RAW) {
                 Photo photo = source as Photo;
@@ -205,8 +215,11 @@ private class BasicProperties : Properti
             
             photo_count = event_photo_count;
             video_count = event_video_count;
+#if CAMERA
         } else if (source is VideoSource || source is VideoImportSource) {
             if (source is VideoSource) {
+#else
+        } else if (source is VideoSource) {
                 Video video = (Video) source;
                 clip_duration = video.get_clip_duration();
 
@@ -214,9 +227,11 @@ private class BasicProperties : Properti
                     dimensions = video.get_frame_dimensions();
 
                 start_time = video.get_exposure_time();
+#if CAMERA
             } else {
                 start_time = ((VideoImportSource) source).get_exposure_time();
             }
+#endif
             end_time = start_time;
         }
     }
@@ -229,10 +244,15 @@ private class BasicProperties : Properti
         foreach (DataView view in iter) {
             DataSource source = view.get_source();
             
+#if CAMERA
             if (source is PhotoSource || source is PhotoImportSource) {                  
                 time_t exposure_time = (source is PhotoSource) ?
                     ((PhotoSource) source).get_exposure_time() :
                     ((PhotoImportSource) source).get_exposure_time();
+#else
+            if (source is PhotoSource) {
+                time_t exposure_time = ((PhotoSource) source).get_exposure_time();
+#endif
 
                 if (exposure_time != 0) {
                     if (start_time == 0 || exposure_time < start_time)
@@ -268,10 +288,15 @@ private class BasicProperties : Properti
                 photo_count += event_photo_count;
                 video_count += event_video_count;
                 event_count++;
+#if CAMERA
             } else if (source is VideoSource || source is VideoImportSource) {
                 time_t exposure_time = (source is VideoSource) ?
                     ((VideoSource) source).get_exposure_time() :
                     ((VideoImportSource) source).get_exposure_time();
+#else
+            } else if (source is VideoSource) {
+                time_t exposure_time = ((VideoSource) source).get_exposure_time();
+#endif
 
                 if (exposure_time != 0) {
                     if (start_time == 0 || exposure_time < start_time)
--- origsrc/shotwell-0.12.3/src/SearchFilter.vala	2012-05-08 16:46:35.000000000 -0500
+++ src/shotwell-0.12.3/src/SearchFilter.vala	2012-11-01 22:23:52.141353300 -0500
@@ -502,8 +502,10 @@ public class SearchFilterActions {
             Core.ViewTracker? tracker = old_tracked_page.get_view_tracker();
             if (tracker is MediaViewTracker)
                 tracker.updated.disconnect(on_media_tracker_updated);
+#if CAMERA
             else if (tracker is CameraViewTracker)
                 tracker.updated.disconnect(on_camera_tracker_updated);
+#endif
         }
         
         CheckerboardPage? new_tracked_page = new_page as CheckerboardPage;
@@ -516,11 +518,13 @@ public class SearchFilterActions {
                 on_media_tracker_updated(tracker);
                 
                 return;
+#if CAMERA
             } else if (tracker is CameraViewTracker) {
                 tracker.updated.connect(on_camera_tracker_updated);
                 on_camera_tracker_updated(tracker);
                 
                 return;
+#endif
             }
         }
         
@@ -545,6 +549,7 @@ public class SearchFilterActions {
         update_sensitivities();
     }
     
+#if CAMERA
     private void on_camera_tracker_updated(Core.Tracker t) {
         CameraViewTracker tracker = (CameraViewTracker) t;
         
@@ -555,6 +560,7 @@ public class SearchFilterActions {
 
         update_sensitivities();
     }
+#endif
     
     private void update_sensitivities() {
         flagged.set_stock_id(((SearchFilterCriteria.FLAG & criteria) != 0 && has_flagged) ?
--- origsrc/shotwell-0.12.3/src/camera/mk/camera.mk	2012-05-08 12:42:59.000000000 -0500
+++ src/shotwell-0.12.3/src/camera/mk/camera.mk	2012-11-01 22:23:52.141353300 -0500
@@ -12,7 +12,6 @@ UNIT_DIR := camera
 # NOTE: Do *not* include the unit's master file, i.e. UNIT_NAME.vala.
 UNIT_FILES := \
 	Branch.vala \
-	CameraTable.vala \
 	GPhoto.vala \
 	ImportPage.vala
 
--- origsrc/shotwell-0.12.3/src/library/LibraryWindow.vala	2012-05-08 16:46:35.000000000 -0500
+++ src/shotwell-0.12.3/src/library/LibraryWindow.vala	2012-11-01 22:23:52.141353300 -0500
@@ -120,7 +120,9 @@ public class LibraryWindow : AppWindow {
     private Library.FlaggedBranch flagged_branch = new Library.FlaggedBranch();
     private Library.LastImportBranch last_import_branch = new Library.LastImportBranch();
     private Library.ImportQueueBranch import_queue_branch = new Library.ImportQueueBranch();
+#if CAMERA
     private Camera.Branch camera_branch = new Camera.Branch();
+#endif
     private Searches.Branch saved_search_branch = new Searches.Branch();
     private bool page_switching_enabled = true;
     
@@ -180,7 +182,9 @@ public class LibraryWindow : AppWindow {
         sidebar_tree.graft(flagged_branch, SidebarRootPosition.FLAGGED);
         sidebar_tree.graft(last_import_branch, SidebarRootPosition.LAST_IMPORTED);
         sidebar_tree.graft(import_queue_branch, SidebarRootPosition.IMPORT_QUEUE);
+#if CAMERA
         sidebar_tree.graft(camera_branch, SidebarRootPosition.CAMERAS);
+#endif
         sidebar_tree.graft(saved_search_branch, SidebarRootPosition.SAVED_SEARCH);
         
         // create and connect extended properties window
@@ -239,7 +243,9 @@ public class LibraryWindow : AppWindow {
         LibraryMonitorPool.get_instance().monitor_installed.connect(on_library_monitor_installed);
         LibraryMonitorPool.get_instance().monitor_destroyed.connect(on_library_monitor_destroyed);
         
+#if CAMERA
         CameraTable.get_instance().camera_added.connect(on_camera_added);
+#endif
         
         background_progress_bar.set_show_text(true);
         
@@ -270,7 +276,9 @@ public class LibraryWindow : AppWindow {
         LibraryMonitorPool.get_instance().monitor_installed.disconnect(on_library_monitor_installed);
         LibraryMonitorPool.get_instance().monitor_destroyed.disconnect(on_library_monitor_destroyed);
         
+#if CAMERA
         CameraTable.get_instance().camera_added.disconnect(on_camera_added);
+#endif
     }
     
     private void on_library_monitor_installed(LibraryMonitor monitor) {
@@ -1051,6 +1059,7 @@ public class LibraryWindow : AppWindow {
         switch_to_page(import_queue_branch.get_queue_page());
     }
     
+#if CAMERA
     private void on_camera_added(DiscoveredCamera camera) {
         // if this page is for a camera which initialized the app, we want to switch to that page
         if (!initial_camera_uris.contains(camera.uri))
@@ -1081,6 +1090,7 @@ public class LibraryWindow : AppWindow {
             switch_to_page(page);
         }
     }
+#endif
 
     // This should only be called by LibraryWindow and PageStub.
     public void add_to_notebook(Page page) {
@@ -1514,8 +1524,10 @@ public class LibraryWindow : AppWindow {
         
         // convert file: URIs into gphoto disk: URIs
         string alt_uri = null;
+#if CAMERA
         if (uri.has_prefix("file://"))
             alt_uri = CameraTable.get_port_uri(uri.replace("file://", "disk:"));
+#endif
         
         // we only add uris when the notification is called on startup
         if (at_startup) {
--- origsrc/shotwell-0.12.3/src/library/mk/library.mk	2012-05-08 16:46:35.000000000 -0500
+++ src/shotwell-0.12.3/src/library/mk/library.mk	2012-11-01 22:23:52.151353400 -0500
@@ -40,7 +40,6 @@ UNIT_USES := \
 	Sidebar \
 	Events \
 	Tags \
-	Camera \
 	Searches \
 	Faces \
 	DataImports
--- origsrc/shotwell-0.12.3/units.mk	2012-05-08 16:46:35.000000000 -0500
+++ src/shotwell-0.12.3/units.mk	2012-11-01 22:24:01.231365900 -0500
@@ -23,7 +23,6 @@ UNITS = \
 	events \
 	tags \
 	faces \
-	camera \
 	searches \
 	config \
 	data_imports
