--- origsrc/shotwell-0.14.1/Makefile	2013-07-30 01:07:10.224488100 -0500
+++ src/shotwell-0.14.1/Makefile	2013-07-30 01:00:40.000000000 -0500
@@ -300,7 +300,6 @@ EXT_PKGS = \
 	gstreamer-base-1.0 \
 	gstreamer-pbutils-1.0 \
 	gtk+-3.0 \
-	gudev-1.0 \
 	libexif \
 	libgphoto2 \
 	json-glib-1.0 \
@@ -332,7 +331,6 @@ EXT_PKG_VERSIONS = \
 	gstreamer-plugins-base-1.0 >= 1.0.0 \
 	gstreamer-pbutils-1.0 >= 1.0.0 \
 	gtk+-3.0 >= 3.0.11 \
-	gudev-1.0 >= 145 \
 	libexif >= 0.6.16 \
 	libgphoto2 >= 2.4.2 \
 	libraw >= 0.13.2 \
--- origsrc/shotwell-0.14.1/src/BatchImport.vala	2013-07-30 01:07:10.233488600 -0500
+++ src/shotwell-0.14.1/src/BatchImport.vala	2013-07-30 01:00:40.000000000 -0500
@@ -121,8 +121,10 @@ public enum ImportResult {
                 return ImportResult.FILE_WRITE_ERROR;
             else
                 return ImportResult.FILE_ERROR;
+#if CAMERA
         } else if (err is GPhotoError) {
             return ImportResult.CAMERA_ERROR;
+#endif
         }
         
         return default_result;
--- origsrc/shotwell-0.14.1/src/Properties.vala	2013-07-30 01:07:10.241489100 -0500
+++ src/shotwell-0.14.1/src/Properties.vala	2013-07-30 01:03:29.670873200 -0500
@@ -194,6 +194,7 @@ private class BasicProperties : Properti
 
         title = source.get_name();
         
+#if CAMERA
         if (source is PhotoSource || source is PhotoImportSource) {           
             start_time = (source is PhotoSource) ? ((PhotoSource) source).get_exposure_time() :
                 ((PhotoImportSource) source).get_exposure_time();
@@ -201,6 +202,13 @@ private class BasicProperties : Properti
                         
             PhotoMetadata? metadata = (source is PhotoSource) ? ((PhotoSource) source).get_metadata() :
                 ((PhotoImportSource) source).get_metadata();
+#else
+        if (source is PhotoSource) {
+            start_time = ((PhotoSource) source).get_exposure_time();
+            end_time = start_time;
+                        
+            PhotoMetadata? metadata = ((PhotoSource) source).get_metadata();
+#endif
 
             if (metadata != null) {
                 exposure = metadata.get_exposure_string();
@@ -220,8 +228,10 @@ private class BasicProperties : Properti
                     Dimensions(0, 0);
             }
             
+#if CAMERA
             if (source is PhotoSource)
                 dimensions = ((PhotoSource) source).get_dimensions();
+#endif
             
             if (source is Photo && ((Photo) source).get_master_file_format() == PhotoFileFormat.RAW) {
                 Photo photo = source as Photo;
@@ -241,8 +251,11 @@ private class BasicProperties : Properti
             
             photo_count = event_photo_count;
             video_count = event_video_count;
+#if CAMERA
         } else if (source is VideoSource || source is VideoImportSource) {
             if (source is VideoSource) {
+#else
+        } else if (source is VideoSource) {
                 Video video = (Video) source;
                 clip_duration = video.get_clip_duration();
 
@@ -250,9 +263,11 @@ private class BasicProperties : Properti
                     dimensions = video.get_frame_dimensions();
 
                 start_time = video.get_exposure_time();
+#if CAMERA
             } else {
                 start_time = ((VideoImportSource) source).get_exposure_time();
             }
+#endif
             end_time = start_time;
         }
     }
@@ -265,10 +280,15 @@ private class BasicProperties : Properti
         foreach (DataView view in iter) {
             DataSource source = view.get_source();
             
+#if CAMERA
             if (source is PhotoSource || source is PhotoImportSource) {                  
                 time_t exposure_time = (source is PhotoSource) ?
                     ((PhotoSource) source).get_exposure_time() :
                     ((PhotoImportSource) source).get_exposure_time();
+#else
+            if (source is PhotoSource) {
+                time_t exposure_time = ((PhotoSource) source).get_exposure_time();
+#endif
 
                 if (exposure_time != 0) {
                     if (start_time == 0 || exposure_time < start_time)
@@ -304,10 +324,15 @@ private class BasicProperties : Properti
                 photo_count += event_photo_count;
                 video_count += event_video_count;
                 event_count++;
+#if CAMERA
             } else if (source is VideoSource || source is VideoImportSource) {
                 time_t exposure_time = (source is VideoSource) ?
                     ((VideoSource) source).get_exposure_time() :
                     ((VideoImportSource) source).get_exposure_time();
+#else
+            } else if (source is VideoSource) {
+                time_t exposure_time = ((VideoSource) source).get_exposure_time();
+#endif
 
                 if (exposure_time != 0) {
                     if (start_time == 0 || exposure_time < start_time)
@@ -521,7 +546,11 @@ private class ExtendedPropertiesWindow :
             if (source == null)
                 return;
             
+#if CAMERA
             if (source is PhotoSource || source is PhotoImportSource) {
+#else
+            if (source is PhotoSource) {
+#endif
                 MediaSource media = (MediaSource) source;
                 file_path = media.get_master_file().get_path();
                 development_path = media.get_file().get_path();
--- origsrc/shotwell-0.14.1/src/SearchFilter.vala	2013-07-30 01:07:10.283491500 -0500
+++ src/shotwell-0.14.1/src/SearchFilter.vala	2013-07-30 01:00:40.000000000 -0500
@@ -492,8 +492,10 @@ public class SearchFilterActions {
             Core.ViewTracker? tracker = old_tracked_page.get_view_tracker();
             if (tracker is MediaViewTracker)
                 tracker.updated.disconnect(on_media_tracker_updated);
+#if CAMERA
             else if (tracker is CameraViewTracker)
                 tracker.updated.disconnect(on_camera_tracker_updated);
+#endif
         }
         
         CheckerboardPage? new_tracked_page = new_page as CheckerboardPage;
@@ -506,11 +508,13 @@ public class SearchFilterActions {
                 on_media_tracker_updated(tracker);
                 
                 return;
+#if CAMERA
             } else if (tracker is CameraViewTracker) {
                 tracker.updated.connect(on_camera_tracker_updated);
                 on_camera_tracker_updated(tracker);
                 
                 return;
+#endif
             }
         }
         
@@ -535,6 +539,7 @@ public class SearchFilterActions {
         update_sensitivities();
     }
     
+#if CAMERA
     private void on_camera_tracker_updated(Core.Tracker t) {
         CameraViewTracker tracker = (CameraViewTracker) t;
         
@@ -545,6 +550,7 @@ public class SearchFilterActions {
 
         update_sensitivities();
     }
+#endif
     
     private void update_sensitivities() {
         flagged.set_stock_id(((SearchFilterCriteria.FLAG & criteria) != 0 && has_flagged) ?
--- origsrc/shotwell-0.14.1/src/camera/mk/camera.mk	2013-07-30 01:07:10.309493000 -0500
+++ src/shotwell-0.14.1/src/camera/mk/camera.mk	2013-07-30 01:00:40.000000000 -0500
@@ -12,7 +12,6 @@ UNIT_DIR := camera
 # NOTE: Do *not* include the unit's master file, i.e. UNIT_NAME.vala.
 UNIT_FILES := \
 	Branch.vala \
-	CameraTable.vala \
 	GPhoto.vala \
 	ImportPage.vala
 
--- origsrc/shotwell-0.14.1/src/library/LibraryWindow.vala	2013-07-30 01:07:10.334494400 -0500
+++ src/shotwell-0.14.1/src/library/LibraryWindow.vala	2013-07-30 01:00:40.000000000 -0500
@@ -120,7 +120,9 @@ public class LibraryWindow : AppWindow {
     private Library.FlaggedBranch flagged_branch = new Library.FlaggedBranch();
     private Library.LastImportBranch last_import_branch = new Library.LastImportBranch();
     private Library.ImportQueueBranch import_queue_branch = new Library.ImportQueueBranch();
+#if CAMERA
     private Camera.Branch camera_branch = new Camera.Branch();
+#endif
     private Searches.Branch saved_search_branch = new Searches.Branch();
     private bool page_switching_enabled = true;
     
@@ -178,7 +180,9 @@ public class LibraryWindow : AppWindow {
         sidebar_tree.graft(flagged_branch, SidebarRootPosition.FLAGGED);
         sidebar_tree.graft(last_import_branch, SidebarRootPosition.LAST_IMPORTED);
         sidebar_tree.graft(import_queue_branch, SidebarRootPosition.IMPORT_QUEUE);
+#if CAMERA
         sidebar_tree.graft(camera_branch, SidebarRootPosition.CAMERAS);
+#endif
         sidebar_tree.graft(saved_search_branch, SidebarRootPosition.SAVED_SEARCH);
         
         // create and connect extended properties window
@@ -237,7 +241,9 @@ public class LibraryWindow : AppWindow {
         LibraryMonitorPool.get_instance().monitor_installed.connect(on_library_monitor_installed);
         LibraryMonitorPool.get_instance().monitor_destroyed.connect(on_library_monitor_destroyed);
         
+#if CAMERA
         CameraTable.get_instance().camera_added.connect(on_camera_added);
+#endif
         
         background_progress_bar.set_show_text(true);
         
@@ -268,7 +274,9 @@ public class LibraryWindow : AppWindow {
         LibraryMonitorPool.get_instance().monitor_installed.disconnect(on_library_monitor_installed);
         LibraryMonitorPool.get_instance().monitor_destroyed.disconnect(on_library_monitor_destroyed);
         
+#if CAMERA
         CameraTable.get_instance().camera_added.disconnect(on_camera_added);
+#endif
     }
     
     private void on_library_monitor_installed(LibraryMonitor monitor) {
@@ -1069,6 +1077,7 @@ public class LibraryWindow : AppWindow {
         switch_to_page(import_queue_branch.get_queue_page());
     }
     
+#if CAMERA
     private void on_camera_added(DiscoveredCamera camera) {
         Camera.SidebarEntry? entry = camera_branch.get_entry_for_camera(camera);
         if (entry == null)
@@ -1095,6 +1104,7 @@ public class LibraryWindow : AppWindow {
             switch_to_page(page);
         }
     }
+#endif
 
     // This should only be called by LibraryWindow and PageStub.
     public void add_to_notebook(Page page) {
@@ -1553,8 +1563,10 @@ public class LibraryWindow : AppWindow {
         
         // convert file: URIs into gphoto disk: URIs
         string alt_uri = null;
+#if CAMERA
         if (uri.has_prefix("file://"))
             alt_uri = CameraTable.get_port_uri(uri.replace("file://", "disk:"));
+#endif
         
         // we only add uris when the notification is called on startup
         if (at_startup) {
--- origsrc/shotwell-0.14.1/src/library/mk/library.mk	2013-07-30 01:07:10.362496000 -0500
+++ src/shotwell-0.14.1/src/library/mk/library.mk	2013-07-30 01:00:40.000000000 -0500
@@ -40,7 +40,6 @@ UNIT_USES := \
 	Sidebar \
 	Events \
 	Tags \
-	Camera \
 	Searches \
 	DataImports \
 	Folders
--- origsrc/shotwell-0.14.1/units.mk	2013-07-30 01:07:10.391497700 -0500
+++ src/shotwell-0.14.1/units.mk	2013-07-30 01:00:40.000000000 -0500
@@ -22,7 +22,6 @@ UNITS = \
 	sidebar \
 	events \
 	tags \
-	camera \
 	searches \
 	config \
 	data_imports \
