--- origsrc/shotwell-0.8.1/Makefile	2011-01-10 17:13:03.000000000 -0600
+++ src/shotwell-0.8.1/Makefile	2011-02-02 01:29:51.522142200 -0600
@@ -62,7 +62,6 @@ SRC_FILES = \
 	DataObject.vala \
 	DataCollection.vala \
 	LibraryWindow.vala \
-	CameraTable.vala \
 	DirectWindow.vala \
 	Properties.vala \
 	CustomComponents.vala \
@@ -280,7 +279,6 @@ EXT_PKGS = \
 	gstreamer-base-0.10 \
 	gtk+-2.0 \
 	glib-2.0 \
-	gudev-1.0 \
 	json-glib-1.0 \
 	libexif \
 	libgphoto2 \
@@ -306,7 +304,6 @@ EXT_PKG_VERSIONS = \
 	glib-2.0 >= 2.24.0 \
 	gstreamer-0.10 >= 0.10.28 \
 	gstreamer-base-0.10 >= 0.10.28 \
-	gudev-1.0 >= 145 \
 	json-glib-1.0 >= 0.7.6 \
 	libexif >= 0.6.16 \
 	libgphoto2 >= 2.4.2 \
--- origsrc/shotwell-0.8.1/src/LibraryWindow.vala	2011-01-10 17:13:03.000000000 -0600
+++ src/shotwell-0.8.1/src/LibraryWindow.vala	2011-02-02 01:33:11.361602500 -0600
@@ -245,6 +245,7 @@ public class LibraryWindow : AppWindow {
         Gtk.drag_dest_set(this, Gtk.DestDefaults.ALL, DEST_TARGET_ENTRIES,
             Gdk.DragAction.COPY | Gdk.DragAction.LINK | Gdk.DragAction.ASK);
         
+#if GUDEV
         // monitor the camera table for additions and removals
         CameraTable.get_instance().camera_added.connect(add_camera_page);
         CameraTable.get_instance().camera_removed.connect(remove_camera_page);
@@ -252,6 +253,7 @@ public class LibraryWindow : AppWindow {
         // need to populate pages with what's known now by the camera table
         foreach (DiscoveredCamera camera in CameraTable.get_instance().get_cameras())
             add_camera_page(camera);
+#endif
         
         // connect to sidebar signal used ommited on drag-and-drop orerations
         sidebar.drop_received.connect(drop_received);
@@ -289,8 +291,10 @@ public class LibraryWindow : AppWindow {
         Tag.global.contents_altered.disconnect(on_tags_added_removed);
         Tag.global.items_altered.disconnect(on_tags_altered);
         
+#if GUDEV
         CameraTable.get_instance().camera_added.disconnect(add_camera_page);
         CameraTable.get_instance().camera_removed.disconnect(remove_camera_page);
+#endif
         
         unsubscribe_from_basic_information(get_current_page());
 
@@ -1335,6 +1339,7 @@ public class LibraryWindow : AppWindow {
         }
     }
     
+#if GUDEV
     private void add_camera_page(DiscoveredCamera camera) {
         ImportPage page = new ImportPage(camera.gcamera, camera.uri, camera.display_name);
 
@@ -1386,6 +1391,7 @@ public class LibraryWindow : AppWindow {
             cameras_marker = null;
         }
     }
+#endif
     
     private PageLayout? get_page_layout(Page page) {
         return page_layouts.get(page);
@@ -1974,8 +1980,10 @@ public class LibraryWindow : AppWindow {
         
         // convert file: URIs into gphoto disk: URIs
         string alt_uri = null;
+#if GUDEV
         if (uri.has_prefix("file://"))
             alt_uri = CameraTable.get_port_uri(uri.replace("file://", "disk:"));
+#endif
         
         // we only add uris when the notification is called on startup
         if (at_startup) {
